### PR TITLE
Migrate to Python 3 and Odoo 11

### PR DESCRIPTION
**Pull request by TDu**

### CODE REVIEW

File: anthem/lyrics/loaders.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 # Copyright 2016-2017 Camptocamp SA
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
 
+from past.builtins import basestring
```
Comment (guewen) [inline anthem/lyrics/loaders.py]: Shouldn't the lib be added in setup.py?

File: anthem/cli.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def _build_odoo_env(self, odoo_args):
         # Restore the default one.
         signal.signal(signal.SIGINT, signal.default_int_handler)
 
-        registry = odoo.modules.registry.RegistryManager.get(dbname)
+        registry = odoo.modules.registry.Registry(dbname)
```
[the remark this answers is not in the document]
Comment (TDu) [reply anthem/cli.py]: Yes I forgot about v9 compatibility.

File: anthem/lyrics/loaders.py (hunk 1 above)
Comment (TDu) [reply anthem/lyrics/loaders.py]: Seems that we need future to have past !?
Added to setup.py.

Comment (guewen) [reply anthem/lyrics/loaders.py]: Yes! :sweat_smile:

**Comment by guewen** [comment]
In py3 the output is a bit too loquacious in this strange way of including b'':

```
root@84e145672c8d:/opt# anthem --interactive songs.install.pre::setup_company
2017-10-27 09:31:59,673 61 INFO odoodb root: Generating grammar tables from /usr/lib/python3.5/lib2to3/Grammar.txt
2017-10-27 09:31:59,702 61 INFO odoodb root: Generating grammar tables from /usr/lib/python3.5/lib2to3/PatternGrammar.txt
b'Setup company...'
b'Setup company: 0.087s'
Welcome to the anthem version 0.10.0 interactive console!
You can use an anthem context as ctx. (ctx.env is an Odoo
environment). If you terminate your session with Ctrl-D,
the database cursor ctx.env.cr will be committed.
Otherwise if you exit() or you raise an exception, the cursor will
be closed without a commit.
>>> import sys
>>> sys.version
'3.5.3 (default, Jan 19 2017, 14:11:04) \n[GCC 6.3.0 20170118]'
>>>
```

**Comment by guewen** [comment]
In the function 
https://github.com/camptocamp/anthem/blob/bc01f5eb21fb93943ee4cb437696df17fb05424e/anthem/output.py#L14-L18
My guess is that for python 3, we just want to print(ustring), it takes care of the rest for us.

**Comment by TDu** [comment]
Build is green again.

File: anthem/lyrics/loaders.py [hunk 3]
```diff
@@ -28,16 +31,22 @@ def load_csv(ctx, model, path, header=None, header_exclude=None, **fmtparams):
 
 
 def csv_unireader(f, encoding="utf-8", **fmtparams):
-    data = csv.reader(
-        codecs.iterencode(codecs.iterdecode(f, encoding), "utf-8"), **fmtparams
-    )
+    if sys.version_info[0] == 3:
+        data = csv.reader(codecs.iterdecode(f, encoding), **fmtparams)
```
Comment (guewen) [inline anthem/lyrics/loaders.py]: Are you confident about this? It seems you are getting bytes out of the file, which is not what we would expect?

Comment (TDu) [reply anthem/lyrics/loaders.py]: Nope, not at all :confused:

File: anthem/lyrics/loaders.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 import codecs
 import csv
```
Comment (simahawk) [inline anthem/lyrics/loaders.py]: using unicodecsv would make this easier to maintain, isn't it?

Comment (guewen) [reply anthem/lyrics/loaders.py]: indeed!

File: README.rst [hunk 5]
```diff
@@ -128,7 +128,19 @@ Odoo 10.0 (Python 2)::
...
+  $ invoke tests.prepare-version 10.0
+  $ OPENERP_SERVER=/tmp/test-anthem-config-10.0.cfg py.test -s tests
+
+Odoo 11.0 (Python 3)::
+
+  $ git clone https://github.com/camptocamp/anthem.git
+  Cloning into 'anthem'...
+  $ cd anthem
+  $ virtualenv -p python3 anthem-env-11.0
+  $ source anthem-env-10.0/bin/activate
```
Comment (guewen) [inline README.rst]: s/10.0/11.0/

File: tasks.py [hunk 6]
```diff
@@ -37,9 +37,9 @@ def tests_prepare(ctx, version):
         ctx.run('wget -nv -c -O odoo.tar.gz {}'.format(url))
         ctx.run('tar xfz odoo.tar.gz')
         ctx.run('rm -f odoo.tar.gz')
-    print('Installing odoo')
+    print('Installing odoo, now')
     ctx.run('pip install -r odoo-{}/requirements.txt -q'.format(version))
-    ctx.run('pip install -e odoo-{} -q'.format(version))
+    ctx.run('pip install -e odoo-{}'.format(version))
```
Comment (guewen) [inline tasks.py]: maybe this one can be reverted?

File: anthem/lyrics/loaders.py (hunk 4 above)
Comment (guewen) [reply anthem/lyrics/loaders.py]: @TDu could you just spend 30 minutes and check if it works with unicodecsv?

File: tasks.py (hunk 6 above)
Comment (TDu) [reply tasks.py]: Done

File: README.rst (hunk 5 above)
Comment (TDu) [reply README.rst]: Done

File: anthem/lyrics/loaders.py (hunk 4 above)
Comment (TDu) [reply anthem/lyrics/loaders.py]: I have made a new PR based on this one to test with unicodecsv https://github.com/camptocamp/anthem/pull/29

**Pull request merged**